root/etc/s6-overlay/s6-rc.d/init-docker-sock-perms/run

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/with-contenv bash
+# shellcheck shell=bash
+
+if [[ -z ${LSIO_NON_ROOT_USER} ]]; then
+    FILES=$(find /var/run/docker.sock -print 2>/dev/null)
+
+    for i in ${FILES}; do
```

Comment:
What does this pattern protect against or solve?

Reply:
Which one?
The env var is explained in the PS section of the PR description. 
For loop is there because it's a standardized script used across various images, fixing perms for various devices and sockets. Linked in the second post.